<h2>Training the Classifier/Filter</h2>
Currently we are trying to train the whole classifier with a single function, the train function. Next is we will call the pipeline to filter and  clean all the tweets we need to clean. After this, we will perform tweet cleaning

<h4>Tweet Cleaning</h4>
This is where the real operation starts. We will filter the tweets based on the saved model, and clean them in a proper manner. The steps are as follows:

* Filter Tweets (filter_pipeline)
* Clean Tweets (data_pipeline)
* format Tweets to pass in our model (data_pipeline)

In [ ]:
from train.classifier import train


train()


 Accuracy:  0.7883597883597884

Classification Report for

               precision    recall  f1-score   support

           0       0.78      0.87      0.83       434
           1       0.80      0.67      0.73       322

    accuracy                           0.79       756
   macro avg       0.79      0.77      0.78       756
weighted avg       0.79      0.79      0.79       756



<h2>Topic Segmentation</h2>
This is where we segment tweets based on the frequency of the word that appears on the document, therefore classifying that word based on the frequency output.

We will perform the following

*  term frequency–inverse document frequency(TF-IDF) to transpose the words into vectors of frequency
* 3 models to test
    * Logistic Regression
    * SGD Classfier
    * SGD Modified Huber
* Check the accuracy metrics 

In [1]:
from train.frequency import train as train_freq

train_freq()

13777

 Accuracy:  0.9412191582002902

Classification Report for

               precision    recall  f1-score   support

           0       0.99      0.96      0.98       201
           1       0.98      0.93      0.95       191
           2       0.99      0.99      0.99       189
           3       0.94      0.91      0.92       181
           4       0.89      0.90      0.90       226
           5       0.96      0.98      0.97       195
           6       0.85      0.92      0.88       195

    accuracy                           0.94      1378
   macro avg       0.94      0.94      0.94      1378
weighted avg       0.94      0.94      0.94      1378


 Accuracy:  0.9412191582002902

Classification Report for

               precision    recall  f1-score   support

           0       0.99      0.96      0.98       201
           1       0.98      0.93      0.96       191
           2       0.99      0.99      0.99       189
           3       0.92      0.91      0.91       181
    

C:\Users\Jayra\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\sklearn\metrics\_classification.py:1308: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\Jayra\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\sklearn\metrics\_classification.py:1308: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\Jayra\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\sklearn\metrics\_classification.py:1308: Unde

<h4>Filipino Vocabulary</h4>
Testing English vocab model to Taglish Corpus 

In [3]:
from train.frequency import test_accuracy

test_accuracy()


13816


<h4>Results</h4>
We can see that our results tends to favour the Frequency distribution since our data is controlled, we have set of keywords related to each other so some keywords may be present to the other keyworded tweet, which brought about the good results since our feature vector tends to see this similarity.

<h4>Error Analysis</h4>
We had a few errors before getting about a good result, we divide them according to the process done

Preprocessing
* Lack of Handling the appropriate stopwords for the dataset
* Spellchecker
* Appropriate Root Word stem
* make tweets unique

Feature Vector (LDA)
* Additional N time for transposing and transforming Cohesion Pair
* took 30 mins to train 13,890 unique tweets

Feature Vector (TF-IDF)
* Less transposing done by reshaping the features into the same Y dim
* Used l2 regularization to lessen the complexity of the model
* Used bounded frequency to lessen the occurence of unhandled stop words

Model (Disaster of Not)
* Best model that has a score of 0.7883597883597884, is SVM
* Out of 5 models, worst result kNN with a score of 0.525

Model (Topic Segmentations)
* Best model that has a score of 0.9412191582002902, is tie to both Logistic Regression and SGD


In [ ]:
from train.frequency import test_accuracy
with open("./saved_models/SGDHuberFreq.pkl", 'rb') as file:
    freq_model = pickle.load(file)

print(type(freq_model))
X_testset = Cleaner(
        ds_train=df_tweets_en,
        column="rawContent"
    ).fit_testset()

test_accuracy()

In [ ]:
import numpy as np

arr = np.array([1,2,3,4,5,6])
print(arr.shape)
arr = arr.reshape(3,2)
print(arr)
print(arr.shape)